### PR TITLE
Use clap for arg handling

### PR DESCRIPTION
**Pull request by BuyMyMojo**

### CODE REVIEW

**Comment by BuyMyMojo** [comment]
The main point of this PR was to add clap for args but I added a little more verbosity to the error handling and changed how it prints the final result too.

If you have any perfences for clap implamentation just ask! :)

File: src/main.rs [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-
+use clap::Parser;
 use ssimulacra2::{compute_frame_ssimulacra2, ColorPrimaries, TransferCharacteristic, Xyb};
 use yuvxyb::Rgb;
 
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// Source image
+    #[arg(help = "Origional unmodified image", value_hint = clap::ValueHint::FilePath)]
```
Comment (lu-zero) [inline src/main.rs:9]:
```suggestion
    #[arg(help = "Original unmodified image", value_hint = clap::ValueHint::FilePath)]
```

File: src/main.rs [hunk 2]
```diff
@@ -42,9 +53,12 @@ fn main() {
...
     )
-    .unwrap();
-    let result = compute_frame_ssimulacra2(source_data, distorted_data).unwrap();
-    eprintln!("{:.8}", result);
+    .expect("Failed to process distorted_data into XYB");
+
+    let result = compute_frame_ssimulacra2(source_data, distorted_data)
+        .expect("Failed to calculate ssimulacra2");
+
+    println!("{result:.8}");
```
Comment (lu-zero) [inline src/main.rs:63]:
```suggestion
    eprintln!("{result:.8}");
```

**Review by lu-zero — APPROVED** [review]
Thank you for the PR, I found a couple of nits to address and defer to @shssoichiro for a second OK.

File: src/main.rs (hunk 2 above)
Comment (shssoichiro) [reply src/main.rs:63]: Actually this probably should have been `println`. My original code of `eprintln` doesn't follow standard convention, the result should go to stdout.

**Comment by shssoichiro** [comment]
I intentionally avoided including clap in order to minimize the number of dependencies. However, I've already had users ask for things like y4m support which would require specifying items such as color matrices and transfer characteristics... for which clap would be very helpful. So I'm not opposed to adding it here, since the work is already done.

**Review by shssoichiro — APPROVED** [review]
Thanks!

File: src/main.rs (hunk 1 above)
Comment (BuyMyMojo) [reply src/main.rs]: Whoops 😅

**Comment by BuyMyMojo** [comment]
Glad to have been of help! 👍

**Pull request merged**